src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx

```diff
@@ -50,3 +52,53 @@ export default {
         ...defaultVisualConfig.parameters,
     },
 };
+
+export const MetadataInstanceEditorWithCustomMetadata: StoryObj<typeof MetadataSidebarRedesign> = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+        await canvas.findByRole('heading', { name: 'Metadata' });
+
+        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' });
+        expect(addTemplateButton).toBeInTheDocument();
+        await userEvent.click(addTemplateButton);
+
+        const customMetadataOption = canvas.getByRole('option', { name: 'Custom Metadata' });
+        expect(customMetadataOption).toBeInTheDocument();
+        await userEvent.click(customMetadataOption);
+    },
+};
+
+export const MetadataInstanceEditorWithTemplate: StoryObj<typeof MetadataSidebarRedesign> = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+        await canvas.findByRole('heading', { name: 'Metadata' });
+
+        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' });
+        expect(addTemplateButton).toBeInTheDocument();
+        await userEvent.click(addTemplateButton);
+
+        const customMetadataOption = canvas.getByRole('option', { name: 'My Template' });
+        expect(customMetadataOption).toBeInTheDocument();
+        await userEvent.click(customMetadataOption);
+    },
+};
+
+export const UnsavedChangesModalWhenChoosingDifferentTemplate: StoryObj<typeof MetadataSidebarRedesign> = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+        await canvas.findByRole('heading', { name: 'Metadata' });
+
+        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' });
+        expect(addTemplateButton).toBeInTheDocument();
+        await userEvent.click(addTemplateButton);
+
+        const customMetadataOption = canvas.getByRole('option', { name: 'Custom Metadata' });
+        expect(customMetadataOption).toBeInTheDocument();
+        await userEvent.click(customMetadataOption);
+
+        await userEvent.click(addTemplateButton);
+        const myTemplateOption = canvas.getByRole('option', { name: 'My Template' });
+        expect(myTemplateOption).toBeInTheDocument();
+        await userEvent.click(myTemplateOption);
```

Comment:
Let's add `expect` to all of these visual tests and check if proper components are rendered :) WDYT?

Reply:
I added to two of them for now. Will work on that.

Reply:
and added the third one